misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JDIEventProcessor.java

```diff
@@ -103,11 +104,31 @@ private boolean processEvent(EventSet eventSet, Event event) {
             configureUserBreakPoints(evt.referenceType());
             eventSet.resume();
         } else if (event instanceof BreakpointEvent) {
-            StoppedEventArguments stoppedEventArguments = new StoppedEventArguments();
-            stoppedEventArguments.setReason(StoppedEventArgumentsReason.BREAKPOINT);
-            stoppedEventArguments.setThreadId(((BreakpointEvent) event).thread().uniqueID());
-            stoppedEventArguments.setAllThreadsStopped(true);
-            context.getClient().stopped(stoppedEventArguments);
+            Value value = null;
+            String qualifiedClassName =
+                getQualifiedClassName(context, ((BreakpointEvent) event).location().declaringType());
+            Map<Integer, BalBreakpoint> balBreakpoints = this.balBreakpoints.get(qualifiedClassName);
+            int lineNumber = ((BreakpointEvent) event).location().lineNumber();
+
+            if (balBreakpoints != null && balBreakpoints.containsKey(lineNumber)) {
```

Comment:
Is this null check for `balBreakpoints` necessary in here?

Reply:
There are instances where `balBreakpoints` can be null. In that case `balBreakpoints.containsKey(lineNumber)` throwing a NPE. Therefore this check is required.

Reply:
I'm still not sure how it can be null, given that its declared as final and already initialized as an empty map :slightly_smiling_face:

Reply:
According to the recent,
`Map<Integer, SourceBreakpoint> sourceBreakpoints = this.sourceBreakpoints.get(qualifiedClassName);` 

Here `sourceBreakpoints` can be null when `this.sourceBreakpoints` doesn't contain any values for key `qualifiedClassName`

Reply:
Sorry I've missed it since you've used the same variable name locally. +1